--- a/benchspec/CPU/500.perlbench_r/src/mg.c
+++ b/benchspec/CPU/500.perlbench_r/src/mg.c
@@ -2057,7 +2057,7 @@ Perl_magic_setdbline(pTHX_ SV *sv, MAGIC *mg)
 }
 
 int
-Perl_magic_getarylen(pTHX_ SV *sv, const MAGIC *mg)
+Perl_magic_getarylen(pTHX_ SV *sv, MAGIC *mg)
 {
     AV * const obj = MUTABLE_AV(mg->mg_obj);
 
@@ -3395,7 +3395,7 @@ Perl_sighandler(int sig)
 
 
 static void
-S_restore_magic(pTHX_ const void *p)
+S_restore_magic(pTHX_ void *p)
 {
     MGS* const mgs = SSPTR(PTR2IV(p), MGS*);
     SV* const sv = mgs->mgs_sv;
--- a/benchspec/CPU/500.perlbench_r/src/perly.c
+++ b/benchspec/CPU/500.perlbench_r/src/perly.c
@@ -191,8 +191,9 @@ do {					\
  * parse stack, thus avoiding leaks if we die  */
 
 static void
-S_clear_yystack(pTHX_  const yy_parser *parser)
+S_clear_yystack(pTHX_ void *p)
 {
+    yy_parser *parser = p;
     yy_stack_frame *ps     = parser->ps;
     int i = 0;
 
--- a/benchspec/CPU/500.perlbench_r/src/proto.h
+++ b/benchspec/CPU/500.perlbench_r/src/proto.h
@@ -2376,7 +2376,7 @@ PERL_CALLCONV int	Perl_magic_get(pTHX_ SV* sv, MAGIC* mg)
 #define PERL_ARGS_ASSERT_MAGIC_GET	\
 	assert(sv); assert(mg)
 
-PERL_CALLCONV int	Perl_magic_getarylen(pTHX_ SV* sv, const MAGIC* mg)
+PERL_CALLCONV int	Perl_magic_getarylen(pTHX_ SV* sv, MAGIC* mg)
 			__attribute__nonnull__(pTHX_1)
 			__attribute__nonnull__(pTHX_2);
 #define PERL_ARGS_ASSERT_MAGIC_GETARYLEN	\
@@ -6232,7 +6232,7 @@ STATIC int	S_magic_methpack(pTHX_ SV *sv, const MAGIC *mg, SV *meth)
 #define PERL_ARGS_ASSERT_MAGIC_METHPACK	\
 	assert(sv); assert(mg); assert(meth)
 
-STATIC void	S_restore_magic(pTHX_ const void *p);
+STATIC void	S_restore_magic(pTHX_ void *p);
 STATIC void	S_save_magic_flags(pTHX_ I32 mgs_ix, SV *sv, U32 flags)
 			__attribute__nonnull__(pTHX_2);
 #define PERL_ARGS_ASSERT_SAVE_MAGIC_FLAGS	\
--- a/benchspec/CPU/500.perlbench_r/src/regcomp.c
+++ b/benchspec/CPU/500.perlbench_r/src/regcomp.c
@@ -3739,7 +3739,7 @@ S_join_exact(pTHX_ RExC_state_t *pRExC_state, regnode *scan,
 
 
 static void
-S_unwind_scan_frames(pTHX_ const void *p)
+S_unwind_scan_frames(pTHX_ void *p)
 {
     scan_frame *f= (scan_frame *)p;
     do {
