--- a/benchspec/CPU/520.omnetpp_r/src/simulator/sectionbasedconfig.cc
+++ b/benchspec/CPU/520.omnetpp_r/src/simulator/sectionbasedconfig.cc
@@ -1020,8 +1020,8 @@ std::vector<const char *> SectionBasedConfiguration::getMatchingConfigKeys(const
 
 const char *SectionBasedConfiguration::getParameterValue(const char *moduleFullPath, const char *paramName, bool hasDefaultValue) const
 {
-    const SectionBasedConfiguration::KeyValue2& entry = (KeyValue2&) getParameterEntry(moduleFullPath, paramName, hasDefaultValue);
-    return entry.getKey()==NULL ? NULL : entry.value.c_str();
+    const SectionBasedConfiguration::KeyValue& entry = getParameterEntry(moduleFullPath, paramName, hasDefaultValue);
+    return entry.getKey()==NULL ? NULL : entry.getValue();
 }
 
 const cConfiguration::KeyValue& SectionBasedConfiguration::getParameterEntry(const char *moduleFullPath, const char *paramName, bool hasDefaultValue) const
@@ -1065,8 +1065,8 @@ std::vector<const char *> SectionBasedConfiguration::getParameterKeyValuePairs()
 
 const char *SectionBasedConfiguration::getPerObjectConfigValue(const char *objectFullPath, const char *keySuffix) const
 {
-    const SectionBasedConfiguration::KeyValue2& entry = (KeyValue2&) getPerObjectConfigEntry(objectFullPath, keySuffix);
-    return entry.getKey()==NULL ? NULL : entry.value.c_str();
+    const cConfiguration::KeyValue& entry = getPerObjectConfigEntry(objectFullPath, keySuffix);
+    return entry.getKey()==NULL ? NULL : entry.getValue();
 }
 
 const cConfiguration::KeyValue& SectionBasedConfiguration::getPerObjectConfigEntry(const char *objectFullPath, const char *keySuffix) const
